### PR TITLE
Refactored CoconsensusModule

### LINKED ISSUE #211: Set the `COCONSENSUS` private constant and create CoconsensusModule::getCoconsensus()
1. In `CoconsensusModule` contract 
    - Set the `coconsensus` 
        - to a `private` constant variable
        - rename it to `COCONSENSUS`
        - value can be determined from [Genesis file generation PM document](https://github.com/mosaicdao/mosaic-pm/blob/master/specification-discussions/20200108-genesis-file-generation.md)
    - Define a `public` function `getCoconsensus` to retrieve the value of the constant `COCONSENSUS`
2. For testing purpose, override the `getCoconsensus` method in test contracts
3. Remove `_coConsensus` argument from `setup` function of `ConsensusCogateway` contract
4. Remove `setupCoConsensus` function from `CoconsensusModule` contract as `_coConsensus` argument is not needed
5. Remove  the `setup` function from `Coreputation` contract as `_coConsensus` argument is not needed
6. Check for other auxilary chain contracts and remove other instances where the `coconsensus` is being setup
    Reference: https://github.com/mosaicdao/mosaic-1/pull/204/files#diff-a30ea18615a491253df579ee2d71e924R64

### PR DESCRIPTION
**Pull request by deepesh-kn**
This pull request contains the following implementation:
1. Changed the term `CoConsensus` to `Coconsensus`.
2. Refactored `CoconsensusModule` to use constant address.
3. Updated all the files to accommodate the above changes accordingly.

fixes: #211

### CODE REVIEW

File: contracts/consensus/CoconsensusModule.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./CoconsensusI.sol";
+
+contract CoconsensusModule {
+
+    /** Address of Coconsensus contract on auxiliary chain. */
+    address private  coconsensusAddress = address(0x0000000000000000000000000000000000004D00);
```
Comment (gulshanvasnani) [inline contracts/consensus/CoconsensusModule.sol]:
```suggestion
    address private constant  coconsensusAddress = address(0x0000000000000000000000000000000000004D00);
```

File: contracts/consensus/CoconsensusModule.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./CoconsensusI.sol";
+
+contract CoconsensusModule {
+
```
Comment (gulshanvasnani) [inline contracts/consensus/CoconsensusModule.sol:20]: ~/* Constants */~

File: contracts/consensus-gateway/ConsensusCogateway.sol [hunk 3]
```diff
@@ -39,7 +39,7 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
      * @notice It sets up consensus cogateway. It can only be called once.
      *
      * @param _metachainId Metachain id of a metablock.
-     * @param _coConsensus Address of coConsensus contract.
+     * @param _coconsensus Address of coconsensus contract.
```
Comment (jayeshbairagi) [inline contracts/consensus-gateway/ConsensusCogateway.sol]: Rename `coconsensus` to `Coconsensus `

File: test/consensus-gateway/setup_consensus_cogateway.js [hunk 4]
```diff
@@ -19,10 +19,10 @@ const { AccountProvider } = require('../test_lib/utils.js');
 const Utils = require('../test_lib/utils.js');
 const ConsensusGatewayUtils = require('./utils');
 
-const SpyCoConsensus = artifacts.require('SpyCoConsensus');
+const SpyCoconsensus = artifacts.require('SpyCoconsensus');
 const ConsensusCogateway = artifacts.require('ConsensusCogateway');
 
-contract('CoConsensusgateway::setup', (accounts) => {
+contract('Coconsensusgateway::setup', (accounts) => {
```
Comment (jayeshbairagi) [inline test/consensus-gateway/setup_consensus_cogateway.js]: Rename `Coconsensusgateway` to `ConsensusCogateway `

File: contracts/consensus/CoconsensusModule.sol (hunk 1 above)
Comment (gulshanvasnani) [reply contracts/consensus/CoconsensusModule.sol]: Resolving after discussion.

File: contracts/consensus-gateway/ConsensusCogateway.sol (hunk 3 above)
Comment (deepesh-kn) [reply contracts/consensus-gateway/ConsensusCogateway.sol]: done

File: test/consensus-gateway/setup_consensus_cogateway.js (hunk 4 above)
Comment (deepesh-kn) [reply test/consensus-gateway/setup_consensus_cogateway.js]: done

File: contracts/consensus/CoconsensusModule.sol (hunk 2 above)
Comment (gulshanvasnani) [reply contracts/consensus/CoconsensusModule.sol:20]: /* Storage */

File: contracts/consensus/CoconsensusModule.sol (hunk 1 above)
Comment (deepesh-kn) [reply contracts/consensus/CoconsensusModule.sol]: It has been purposely removed. 
With constant function `getCoconsensus()` is `pure`, and when we override it in the tests it has to be a `view` function, the compiler complains about it. 

We are achieving our objective without it being a constant.

**Review by jayeshbairagi — APPROVED** [review]
LGTM

**Pull request merged**